### PR TITLE
fix: remove support for Node.JS 10.x

### LINKED ISSUE #418: Drop Node10, add Node16
Node.JS 10 is officially EOL, and Node 16 has been officially released -- https://nodejs.org/en/about/releases/

We should drop Node10 from our CI and support, and add Node16.

### PR DESCRIPTION
**Pull request by kozlove-aws**
### Problem
Node.js 10 is not supported anymore since April 10, 2021
We need to update it with newer library.

### Solution 
Replaced nodejs10 with nodejs14 in Docker file for lambda layer.
Also was changed to take `amazonlinux` from `public.ecr` for avoiding limit hit.

### Testing
Was build image and deployed on local account.
Was verified that lambda layer is using `nodejs12.x, nodejs14.x`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by ddneilson — CHANGES_REQUESTED** [review]
Looks like Node 10 references here that should also be updated

* https://github.com/aws/aws-rfdk/blob/8a55f32124e2cfbadb33437d2d0494580a9eebac/integ/package.json#L110
* https://github.com/aws/aws-rfdk/blob/8a55f32124e2cfbadb33437d2d0494580a9eebac/packages/aws-rfdk/package.json#L210
* https://github.com/aws/aws-rfdk/blob/8a55f32124e2cfbadb33437d2d0494580a9eebac/tools/pkglint/lib/rules.ts#L632
* https://github.com/aws/aws-rfdk/blob/8a55f32124e2cfbadb33437d2d0494580a9eebac/tools/cdk-build-tools/package.json#L73

**Review by ddneilson — DISMISSED** [review]

File: tools/pkglint/lib/rules.ts [hunk 1]
```diff
@@ -376,11 +376,11 @@ export class NodeCompatibility extends ValidationRule {
 
   public validate(pkg: PackageJson): void {
     const atTypesNode = pkg.getDevDependency('@types/node');
-    if (atTypesNode && !atTypesNode.startsWith('^10.')) {
+    if (atTypesNode && !atTypesNode.startsWith('^12.')) {
       pkg.report({
         ruleName: this.name,
         message: `packages must support node version 10 and up, but ${atTypesNode} is declared`,
```
Comment (horsmand) [inline tools/pkglint/lib/rules.ts]:
```suggestion
        message: `packages must support node version 12 and up, but ${atTypesNode} is declared`,
```

**Review by horsmand — CHANGES_REQUESTED** [review]
Looks good, I only found one spot where we're referencing node 10 instead of node 12 in a message that pkglint prints out. I'm also wondering if we should be running `npx yarn install` so it can update the `yarn.lock` file. I see this line in there:
```
"@types/node" "^10.17.60"
```

File: tools/pkglint/lib/rules.ts (hunk 1 above)
Comment (kozlove-aws) [reply tools/pkglint/lib/rules.ts]: Good catch. `yarn.lock` is updated now.

Comment (horsmand) [reply tools/pkglint/lib/rules.ts]: Did you include your changes in that most recent push? I don't see this rule or the yarn.lock updated.

Comment (kozlove-aws) [reply tools/pkglint/lib/rules.ts]: You are right. Forgot to add files to commit.

**Review by horsmand — CHANGES_REQUESTED** [review]
This looks good, I don't see anything that needs updating, but it seems you have a conflict in your yarn.lock so you'll need to rebase your changes and push them. I'll approve after that.

**Review by horsmand — APPROVED** [review]

**Review by jericht — APPROVED** [review]
Looks good!

I only see the one reference to `"@types/node@^10.17.60"` that @horsmand [mentioned](https://github.com/aws/aws-rfdk/pull/459#pullrequestreview-685709529) in our `yarn.lock` file and it looks like it's a [dependency of `@aws-cdk/cloudformation-diff`](https://github.com/aws/aws-rfdk/blob/18a8098944f3599446c0f319eaf46b383724cfc3/yarn.lock#L780-L791). We can force it to resolve to `@types/node@12.x.x` by adding an entry in the [resolutions in the root `package.json`](https://github.com/aws/aws-rfdk/blob/18a8098944f3599446c0f319eaf46b383724cfc3/package.json#L112-L119):
```
  "resolutions": {
    // ...
    "**/@aws-cdk/cloudformation-diff/**/@types/node": "^12.8.3"
  },
```

These are just the typings though, so not really necessary. I checked out `@aws-cdk/cloudformation-diff`, changed it to depend on `"@types/node": "^12.18.3"`, then built and tested it and it succeeded, so I think it's safe to assume it's not relying on things that are incompatible between Node 10 and 12.

**Review by jusiskin — APPROVED** [review]

**Pull request merged**